tutorials/kube-mqtt/index.md

```diff
@@ -0,0 +1,267 @@
+date_published: 2019-07-01
+---
+
+Preston Holmes | Solution Architect | Google
+
+<!-- diagram sources: https://docs.google.com/presentation/d/1p2srfYUkqnXIKR4zGj099PIKOzz1RRi28eOcgIch1tc/edit#slide=id.p -->
+
+## Introduction
+
+This tutorial demonstrates how to deploy [Mosquitto](https://mosquitto.org/) MQTT broker in [Kubernetes](https://kubernetes.io/) using the broker's bridge feature to map a specific [MQTT](http://mqtt.org/) topic namespace to [Google Cloud IoT Core](https://cloud.google.com/iot-core/).
+
+
+## Setup
+
+### Clone the tutorial Repo
+#### TODO update to prod
+
+	git clone https://github.com/ptone/community.git
+	cd community
+	git checkout kube-mqtt
+	cd tutorials/kube-mqtt
+
+### Prerequisite: Kubernetes Cluster
+
+
+
+It is assumed you have running Kubernetes cluster. This can be via [Minikube](https://kubernetes.io/docs/tutorials/hello-minikube/), [Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine/), etc. For the tutorial, you can also use Kubernetes Engine clusters, however the context of this solution is based on non-cloud clusters in Edge environments.
+
+You will also need a recent `kubectl` command (>=1.14) with [kustomize integration](https://kubernetes.io/blog/2019/03/25/kubernetes-1-14-release-announcement/). If not, install it from [here](https://kubernetes.io/docs/tasks/tools/install-kubectl/). Ensure that Kubectl is configured to talk to your cluster.
+
+### Setting up the cloud environment
+
+1.  Create a project in the [GCP Console][console].
+1.  [Enable billing for your project](https://cloud.google.com/billing/docs/how-to/modify-project).
+1.  Use [Cloud Shell][shell] or install the [Google Cloud SDK][sdk].
+
+
+In this tutorial, you may open multiple shell tabs or sessions. You should set these environment variables in each one:
+
+	# the project should already be set in cloud shell
+	gcloud config set project <project-id>
+	
+	export PROJECT=$(gcloud config list project --format "value(core.project)" )
+	export REGION=us-central1
+	export REGISTRY=gadgets
+	export DEVICE=bridge
+	
+	
+[console]: https://console.cloud.google.com/
+[shell]: https://cloud.google.com/shell/
+[sdk]: https://cloud.google.com/sdk/
+
+Enable APIs:
+	
+	gcloud services enable \
+	cloudiot.googleapis.com \
+	container.googleapis.com \
+	containerregistry.googleapis.com \
+	pubsub.googleapis.com \
+	cloudbuild.googleapis.com
+		
+	
+## Build the bridge manager container
+
+For Mosquitto you will use the default published container from Docker. But you need another container in the deployment which manages the credentials that allow the bridge to connect to Google Cloud.
+
+	cd refresher-container
+	gcloud builds submit --tag gcr.io/$PROJECT/refresher .
+
+## Create Cloud resources:
+
+### Create the device keys
+
+	cd ../bridge
+	openssl ecparam -genkey -name prime256v1 -noout -out ec_private.pem
+	openssl ec -in ec_private.pem -pubout -out ec_public.pem
+
+### Create IoT Core resources
+
```

Comment:
Prefer to avoid starting code section without leading sentence:

The following commands will allocate the IoT Core resources.